cfme/fixtures/cli.py

```diff
@@ -89,11 +92,19 @@ def appliance_with_preset_time(temp_appliance_preconfig_funcscope):
 
 
 @pytest.yield_fixture()
-def ipa_crud(configured_appliance, ipa_creds):
-    configured_appliance.appliance_console_cli.configure_ipa(ipa_creds['ipaserver'],
-        ipa_creds['username'], ipa_creds['password'], ipa_creds['domain'], ipa_creds['realm'])
+def ipa_crud():
+    try:
+        ipa_keys = [key
+                    for key, yaml in iteritems(auth_data.auth_providers)
+                    if yaml.type == authutil.FreeIPAAuthProvider.auth_type]
+        ipa_provider = authutil.get_auth_crud(ipa_keys[0])
+    except AttributeError:
+        pytest.mark.uncollect('Unable to parse auth_data.yaml for freeipa server')
+    except IndexError:
+        pytest.mark.uncollect('No freeipa server available for testing')
+    logger.info('Configuring first available freeipa auth provider %s', ipa_provider)
 
-    yield(configured_appliance)
+    yield(ipa_provider)
```

Comment:
**QUESTION:** Why yield if there is no code to act on after control is returned to the function?

Reply:
just didn't change the original use, can be a return.

Reply:
I don't that we have to change it but it seemed like it being a yield fixture was not necessary.